(ESPSTUDIO-8275)

mtlljm opened the pull request:
Plugin deployment option for different namespaces on Viya

Update readme to reflect that the scripts are now run seperate configure-grafana.sh and then the appropriate register-oauth-client-*.sh

no longer need to provide the whole plugin url, just the version

Change-Id: Iab2ab7cc80ddb6c9abc717f28eddc37a0a2d41c6

mtlljm committed:
(ESPSTUDIO-8275)
Plugin deployment option for different namespaces on Viya

Update readme to reflect that the scripts are now run seperate configure-grafana.sh and then the appropriate register-oauth-client-*.sh

no longer need to provide the whole plugin url, just the version

Change-Id: Iab2ab7cc80ddb6c9abc717f28eddc37a0a2d41c6

chrismwsas commented on README.md:
````diff
@@ -105,17 +103,40 @@ Use the installation script to install the plug-in:
+   - ```
+     bash register-oauth-client-viya.sh <esp-namespace> <grafana-namespace>
+     ```
+7. If you have chosen SAS VIYA as your OAuth provider then the Content security policy (CSP) for SAS Logon needs to be altered to allow the Grafana hostname to be used as a target of form submission. 
````
This step is only required if Grafana is running outside the Viya namespace.

katjam-sas replied:
Here is slightly edited text for step 7. There are a few little changes so you might want to copy and paste. The main changes that it's "the SAS Viya platform" not "SAS VIYA", "hostname" should be "host name", and use a bulleted rather than numbered list when you're presenting alternatives (whereas numbered list implies that that you must perform both steps in the order shown).

If your OAuth provider is the SAS Viya platform and Grafana is not running in the same namespace as the SAS Viya platform, you must update the Content Security Policy (CSP) for SAS Logon to allow the Grafana host name to be used as a target of form submission. If you do not update the CSP, the browser blocks the redirect. You can update the CSP in one of the following two ways:

   - Use SAS Environment Manager to update the _content-security-policy_ value under the _sas.commons.web.security_ section.
   - Update the _sas-logon-app_ deployment to add the _SAS_COMMONS_WEB_SECURITY_CONTENTSECURITYPOLICY_ environment variable.
   Update either SAS Environment Manager or the _sas-logon-app_ deployment with the following value, substituting the Grafana host name:
   ```
   default-src 'self'; style-src 'self'; font-src 'self' data:;
   frame-ancestors 'self'; form-action 'self' <grafana-host>;

katjam-sas commented on README.md:
```diff
@@ -81,14 +81,12 @@ This section is relevant only to internal users at SAS.
 
-An installation script is provided to install the plug-in and configure Grafana. The installation script performs the following tasks:
+Installation scripts are provided to install the plug-in and configure Grafana. These scripts perform the following tasks:
  * Modifies the Grafana deployment by adding the GF_INSTALL_PLUGINS environment variable to enable Grafana to install the plug-in.
```
You changed "An installation script is" to "Installation scripts are", and as a result you now need to adjust the text in the bulleted list too. For example, in the first bullet, change "Modifies" to "Modify".

katjam-sas commented on README.md:
```diff
@@ -105,17 +103,40 @@ Use the installation script to install the plug-in:
+5. Run `configure-grafana.sh`, adjusting the command to specify the following variables:
    - The Kubernetes _namespace_ in which SAS Event Stream Processing is installed.
    - The _version_ of the plug-in that you want to install. Ensure that you specify a version of the plug-in that is compatible with your version of Grafana.
    - The _oauth-provider_ of the environment. Select one of the following options: **uaa**, **keycloak** or **viya**.
```
Change:
**uaa**, **keycloak** or **viya**.
to:
`uaa`, `keycloak` or `viya`.
Bold highlighting would be appropriate if we were selecting things in a UI. I know this is not related to the changes you made today, but figured I'd mentioned now since I noticed it.

katjam-sas commented on README.md:
````diff
@@ -105,17 +103,40 @@ Use the installation script to install the plug-in:
-   bash configure-grafana.sh <namespace> https://github.com/sassoftware/grafana-esp-plugin/download/<version>/sasesp-plugin-<version>.zip <oauth-provider>
+   bash configure-grafana.sh <namespace> <version> <oauth-provider>
+   ```
+6. For your chosen OAuth provider run the appropriate script adjusting the command to specify the following variables:
````
How about: "Run one of the following three scripts, depending on your chosen OAuth provider. Adjust the command to specify the following variables."

katjam-sas commented on README.md:
````diff
@@ -105,17 +103,40 @@ Use the installation script to install the plug-in:
+   - The Kubernetes namespace in which SAS Event Stream Processing is installed, _esp-namespace_.
+   - (Optional) The Kubernetes namespace in which Grafana is installed, _grafana-namespace_ if this differs from the namespace in which SAS Event Stream Processing is installed.
+   - ```
+     bash register-oauth-client-keycloak.sh <esp-namespace> <grafana-namespace>
````
These three commands are each rendered with a bullet in front of them. Can you move them to appear after the bulleted list please (like in step 5)? That is, remove the - character that's currently in front of  ``` for each command.

mtlljm committed:
(ESPSTUDIO-8275)
Plugin deployment option for different namespaces on Viya

Update readme to reflect that the scripts are now run seperate configure-grafana.sh and then the appropriate register-oauth-client-*.sh

no longer need to provide the whole plugin url, just the version

Change-Id: I320ada63fc1428305cb138e2fdabb2f08551b74a

mtlljm committed:
(ESPSTUDIO-8275)
Plugin deployment option for different namespaces on Viya

Update readme to reflect that the scripts are now run seperate configure-grafana.sh and then the appropriate register-oauth-client-*.sh

no longer need to provide the whole plugin url, just the version

Change-Id: If28958acc3af789acf39785c368552f86dbff5c2

mtlljm committed:
(ESPSTUDIO-8275)
Plugin deployment option for different namespaces on Viya

Update readme to reflect that the scripts are now run seperate configure-grafana.sh and then the appropriate register-oauth-client-*.sh

no longer need to provide the whole plugin url, just the version

Change-Id: I0c85a50fe176c226a6aab26611c241b778970eea
